lib/charms/opensearch/v0/opensearch_distro.py

```diff
@@ -452,24 +452,30 @@ def normalize_allocation_exclusions(exclusions: Union[List[str], Set[str], str])
 
         return exclusions
 
-    @staticmethod
-    def missing_sys_requirements() -> List[str]:
+    def missing_sys_requirements(self) -> List[str]:
         """Checks the system requirements."""
+
+        def apply(prop: str, value: str) -> bool:
+            """Apply a sysctl value and check if it was set."""
+            try:
+                self._run_cmd(f"sysctl -w {prop}={value}")
+                return self._run_cmd(f"sysctl -n {prop}") == value
+            except OpenSearchCmdError:
+                return False
+
         missing_requirements = []
 
-        max_map_count = int(subprocess.getoutput("sysctl vm.max_map_count").split("=")[-1].strip())
-        if max_map_count < 262144:
-            missing_requirements.append("vm.max_map_count should be at least 262144")
+        prop, val = "vm.max_map_count", "262144"
+        if self._run_cmd(f"sysctl -n {prop}") < val and not apply(prop, val):
+            missing_requirements.append(f"{prop} should be at least {val}")
 
-        swappiness = int(subprocess.getoutput("sysctl vm.swappiness").split("=")[-1].strip())
-        if swappiness > 0:
-            missing_requirements.append("vm.swappiness should be 0")
+        prop, val = "vm.swappiness", "0"
+        if self._run_cmd(f"sysctl -n {prop}") > val and not apply(prop, val):
+            missing_requirements.append(f"{prop} should be {val}")
 
-        tcp_retries = int(
-            subprocess.getoutput("sysctl net.ipv4.tcp_retries2").split("=")[-1].strip()
-        )
-        if tcp_retries > 5:
-            missing_requirements.append("net.ipv4.tcp_retries2 should be 5")
+        prop, val = "net.ipv4.tcp_retries2", "5"
+        if self._run_cmd(f"sysctl -n {prop}") > val and not apply(prop, val):
+            missing_requirements.append(f"{prop} should be at most {val}")
```

Comment:
What will we do about `fs.file-max`?

Reply:
This is already set in the snap through `ulimits` see [here](https://github.com/canonical/opensearch-snap/blob/main/scripts/wrappers/start.sh#L8) and [here](https://github.com/canonical/opensearch-snap/blob/main/scripts/wrappers/sys/set-sys-config.sh#L29-L32)